src/main.rs

```diff
@@ -190,14 +190,21 @@ fn run() -> Result<()> {
     if let Some(sub_matches) = matches.subcommand_matches("generate") {
         let theme = load_theme(&sub_matches, &user_config_path, color_mode)?;
 
-        let mut filetypes_list = filetypes.mapping.keys().collect::<Vec<_>>();
-        filetypes_list.sort_unstable_by_key(|entry| entry.len());
-
-        let mut ls_colors: Vec<String> = vec![];
-        for filetype in filetypes_list {
-            let category = &filetypes.mapping[filetype];
-            ls_colors.push(format!("{}={}", filetype, theme.get_style(&category)?));
-        }
+        let mut mapping = filetypes
+            .mapping
+            .iter()
+            .map(|(filetype, category)| (filetype, theme.get_style(category)))
+            .map(|(filetype, style)| style.map(|style| (filetype, style)))
+            .collect::<Result<Vec<_>>>()?;
```

Comment:
Minor: Maybe add a type annotation instead of `collect::<T>()` (just like below)
```suggestion
        let mut mapping: Result<Vec<_>> = filetypes
            .mapping
            .iter()
            .map(|(filetype, category)| (filetype, theme.get_style(category)))
            .map(|(filetype, style)| style.map(|style| (filetype, style)))
            .collect()?;
```

Reply:
It would be `mapping: Vec<_> = ...collect()?;`.  But rustc's type inference doesn't actually handle that.  I could do

```rust
let mut mapping: Vec<_> = ...collect::<Result<_>>()?;
```

but I figured I might as well put everything in one place